nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GenerateTableFetch.java

```diff
@@ -87,6 +87,18 @@
             .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
             .build();
 
+    public static final PropertyDescriptor AUTO_INCREMENT_KEY = new PropertyDescriptor.Builder()
+            .name("gen-table-fetch-partition-index")
+            .displayName("AUTO_INCREMENT(index) column name")
+            .description("The column has AUTO_INCREMENT attribute and index."
+                    + "If there is a column with AUTO_INCREMENT property and index in the database, we can use index instead of using OFFSET."
+                    + "The value must start by 1")
+            .defaultValue("null")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .required(true)
```

Comment:
I think this property should be optional, `required(true)` and `defaultValue("null")` can be removed.